apps/hermes/server/src/state/aggregate.rs

```diff
@@ -568,6 +616,145 @@ where
     })
 }
 
+async fn get_verified_twaps_with_update_data<S>(
+    state: &S,
+    price_ids: &[PriceIdentifier],
+    start_time: RequestTime,
+    end_time: RequestTime,
+) -> Result<TwapsWithUpdateData>
+where
+    S: Cache,
+{
+    let mut calculated_twaps = Vec::new();
+    let mut all_update_data = Vec::new();
+
+    for price_id in price_ids {
+        let start_messages = state
+            .fetch_message_states(
+                vec![price_id.to_bytes()],
```

Comment:
why not fetch all price_ids at the same time

Reply:
Initially my concern was that if `len(all_price_id_start_messages)` != `len(all_price_ids_end_messages)`.
basically if there is an unmatched start/end msg for a given price ID. But thinking about it again, since we aren't supporting partial success, i think it's acceptable to just error out in that case. Fetching all the messages up front does match the `fetch_message_states` API better as well, will make that change